tiredrive/strategies.py

```diff
@@ -76,12 +76,45 @@ def turn_forward_left(self):
             self.robot.left_motor.set(0)
             self.robot.right_motor.set(val)
             yield
+"""
+
+
+class TurnStrategy:
+
+    def __init__(self, robot):
+        self.robot = robot
+        self.robot.strategies['turn'] = self
+
+    def autonomousInit(self):
+        self.auto = ParallelGenerators()
+        self.auto.add("turn", self.turn(90))
+        self.auto.add("brake", self.brake(), after="turn")
+        self.auto.add("wait", self.wait(), after="brake")
+
+    def autonomousPeriodic(self):
+        self.auto.next()
+
+    def turn(self, angle):
+        while abs(self.robot.gyro.getAngle()) % 360 < angle:
```

Comment:
wouldn't you be wanting to compare the <longthing>difference between the initial gyro value and the current gyro value</longthing> against angle?